### DISCUSSION TITLE
How to get belongsToMany Relationship without soft deleted records

### DISCUSSION THREAD

**Discussion by nmp4817** [laravel/framework#38171] [Q&A]
we have a DB relationship as follows:

Table **A** 
Table **B**
Table **C**

where C is the relationship table between **A** and **B**

we wrote the relationship as follows inside **Model A**:
`$this->belongsToMany(B::class, 'table_c', 'a_id', 'b_id')`

> the equivalent SQL of which is 

```
select b.*, c.a_id as pivot_a_id, c.b_id as pivot_b_id 
from b
inner join c on c.b_id = b.id
where c.a_id=x and b.deleted = 0 
```
As you can see query do not include soft delete condition in inner join! How to achieve this?

- `deleted` is a soft delete column.

**Comment by dennisprudlo** [comment]
You can filter your relationship by its pivot table https://laravel.com/docs/8.x/eloquent-relationships#filtering-queries-via-intermediate-table-columns

**Reply by nmp4817** [reply]
Hi thanks for replying
can you please give me a statement, how would it look like for the above example?

**Reply by dennisprudlo** [reply]
Can't you use `->wherePivotNull('deleted_at')` or however your soft delete column is called?

**Reply by nmp4817** [reply]
Ohh I see 
I already did this
`$this->belongsToMany(B::class, 'table_c', 'a_id', 'b_id')->where('b.deleted', '=', 0);`

Thanks for introducing me to new where method
